src/code/z_bgcheck.c

```diff
@@ -1,432 +1,4713 @@
+    posTemp = *pos;
+    posTemp.y = *outY;
+    tempY = *outY;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_800C074C.s")
+    if (BgCheck_CheckDynaCeiling(colCtx, xpFlags, &tempY, &posTemp, checkHeight, outPoly, outBgId, actor)) {
+        *outY = tempY;
+        result = true;
+    }
+    return result;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_800C07F0.s")
+/**
+ * Tests for collision with any ceiling poly
+ * `checkHeight` must be a positive value
+ * returns true if a collision occurs, else false
+ * `outY` returns the displaced y coordinate needed to not collide with the poly
+ */
+s32 BgCheck_AnyCheckCeiling(CollisionContext* colCtx, f32* outY, Vec3f* pos, f32 checkHeight) {
+    CollisionPoly* poly;
+    s32 bgId;
+
+    return BgCheck_CheckCeilingImpl(colCtx, COLPOLY_IGNORE_NONE, outY, pos, checkHeight, &poly, &bgId, NULL);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/BgCheck_PolygonCollidesWithSphere.s")
+/**
+ * Tests for collision with any entity solid ceiling poly
+ * `checkHeight` must be a positive value
+ * returns true if a collision occurs, else false
+ * `outY` returns the displaced y coordinate needed to not collide with the poly
+ */
+s32 BgCheck_EntityCheckCeiling(CollisionContext* colCtx, f32* outY, Vec3f* pos, f32 checkHeight,
+                               CollisionPoly** outPoly, s32* outBgId, Actor* actor) {
+    return BgCheck_CheckCeilingImpl(colCtx, COLPOLY_IGNORE_ENTITY, outY, pos, checkHeight, outPoly, outBgId, actor);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/BgCheck_ScenePolygonListsInsertSorted.s")
+#ifdef NON_MATCHING
+/**
+ * Tests if a line from `posA` to `posB` intersects with a poly
+ * returns true if it does, else false
+ * `posB`? `posResult` returns the point of intersection
+ * `outPoly` returns the pointer to the intersected poly, while `outBgId` returns the entity the poly belongs to
+ */
+s32 BgCheck_CheckLineImpl(CollisionContext* colCtx, u16 xpFlags1, u16 xpFlags2, Vec3f* posA, Vec3f* posB,
+                          Vec3f* posResult, CollisionPoly** outPoly, s32* outBgId, Actor* actor, f32 chkDist,
+                          u32 bccFlags) {
+    StaticLookup* lookupTbl;
+    StaticLookup* iLookup;
+    s32 subdivMin[3];
+    s32 subdivMax[3];
+    s32 i;
+    s32 result;
+    Vec3f posBTemp;
+    StaticLineTest checkLine;
+    Vec3f sectorMin;
+    Vec3f sectorMax;
+    s32 k;
+    StaticLookup* lookup;
+    s32 j;
+    StaticLookup* jLookup;
+    s32 temp_lo;
+    s32 pad[2];
+
+    lookupTbl = colCtx->lookupTbl;
+    posBTemp = *posB;
+
+    *outBgId = BGCHECK_SCENE;
+
+    BgCheck_ResetPolyCheckTbl(&colCtx->polyNodes, colCtx->colHeader->numPolygons);
+    BgCheck_GetStaticLookupIndicesFromPos(colCtx, posA, (Vec3i*)&subdivMin);
+    BgCheck_GetStaticLookupIndicesFromPos(colCtx, &posBTemp, (Vec3i*)&subdivMax);
+    *posResult = *posB;
+    checkLine.outDistSq = 1.0e38f;
```

Comment:
```suggestion
    checkLine.outDistSq = FLT_MAX;
```
Is this float_max? It's on the same order of magnitude

Reply:
FLT_MAX is defined as `340282346638528859811704183484516925440.0f`, which is not `1.0e38f`.

Reply:
Oh, what about `FLT_MAX / 3.4f`?` differences between possible adjacent floats becomes very large at large floats, so there's I think it can be quite lenient on difference in values